### DISCUSSION TITLE
@walletconnect/utils backward-compatible

### DISCUSSION THREAD

**Discussion by khoanguyen-yang** [WalletConnect/walletconnect-monorepo#1565] [Developer Support] [answered]
Hi guys, how to achieve backward-compatibility (between V1 and V2 util helpers) for `@walletconnect/utils`?

We currently planning to support both V1 for and V2 for our wallet.

We want to use the parse URI method of `@walletconnect/utils` but `@walletconnect/utils@1.7.8` and `@walletconnect/utils@2.1.0` have different methods for URI parsing.
- `parseWalletConnectUri` in v1.7.8 for parsing V1 URIs (with `handshakeTopic`, `bridge`, `key`, etc.)
- `parseUri` in v2.1.0 for parsing V2 URIs (with `symKey`, `relay`, etc.)

**Comment by bkrem** [comment, marked as answer]
Hi @khoanguyen-yang,

We provide the `@walletconnect/legacy-...` packages specifically for this purpose, see this section of the docs:
https://docs.walletconnect.com/2.0/javascript/sign/wallet-usage#migrating-from-v1x

You should be able to install `@walletconnect/legacy-utils` (for v1) side by side with the latest `@walletconnect/utils@2.x.x` (for v2) and then use utilities from both.

Hope that helps, please let us know if you run into any issues using the `legacy-` packages! 🙏

**Reply by khoanguyen-yang** [reply]
many thanks!
